app/utils/file/file_picker/index.test.ts

```diff
@@ -0,0 +1,803 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+/* eslint-disable max-lines */
+
+import RNUtils from '@mattermost/rnutils';
+import {applicationName} from 'expo-application';
+import {createIntl} from 'react-intl';
+import {Alert, Platform} from 'react-native';
+import DocumentPicker, {type DocumentPickerResponse} from 'react-native-document-picker';
+import {launchCamera, launchImageLibrary, type Asset, type ImagePickerResponse} from 'react-native-image-picker';
+import Permissions from 'react-native-permissions';
+
+import {getTranslations} from '@i18n';
+import {dismissBottomSheet} from '@screens/navigation';
+import TestHelper from '@test/test_helper';
+import {extractFileInfo, lookupMimeType} from '@utils/file';
+import {logWarning} from '@utils/log';
+
+import FilePickerUtil from '.';
+
+jest.mock('expo-file-system');
+jest.mock('react-native-image-picker');
+
+jest.mock('react-native-document-picker', () => ({
+    pick: jest.fn(async () => []),
+}));
+
+jest.mock('@screens/navigation', () => ({
+    dismissBottomSheet: jest.fn(),
+}));
+
+jest.mock('@utils/file', () => ({
+    extractFileInfo: jest.fn(),
+    lookupMimeType: jest.fn(),
+}));
+
+jest.mock('@utils/log', () => ({
+    logWarning: jest.fn(),
+}));
+
+jest.mock('@mattermost/rnutils', () => ({
+    getRealFilePath: jest.fn(),
+    isRunningInSplitView: jest.fn().mockReturnValue({isSplit: false, isTablet: false}),
+}));
+
+describe('FilePickerUtil', () => {
+    const mockUploadFiles = jest.fn();
+    const intl = createIntl({locale: 'en', messages: getTranslations('en')});
+    const originalSelect = Platform.select;
+
+    let filePickerUtil: FilePickerUtil;
+
+    beforeAll(() => {
+        Platform.select = ({android, ios, default: dft}: any) => {
+            if (Platform.OS === 'android' && android) {
+                return android;
+            } else if (Platform.OS === 'ios' && ios) {
+                return ios;
+            }
+
+            return dft;
+        };
```

Comment:
Since this is repeated a few times, perhaps extract this into a higher scope (1 describe() higher)?